tools/capture_radar.rb

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+require File.expand_path("../config/environment", __dir__)
+require 'trollop'
+
+opts = Trollop.options(ARGV) do
+  banner "USAGE:   #{__FILE__} -h <number of hours back to query metrics>\n" \
+         "Example: #{__FILE__} -d <number of days back to query metrics>"
+
+  opt :hours, "Hours", :short => "h", :type => :int, :default => 4
+  opt :days,  "Days",  :short => "d", :type => :int
+  opt :label, "Label", :short => "l", :type => :string, :default => "com.redhat.component"
+end
+Trollop.die :hours, "is required" unless opts[:hours] || opts[:days_given]
+
+Metric
+class Metric
+  def derived_cpu_total_cores_used
+    return if cpu_usage_rate_average.nil? || derived_vm_numvcpus.nil? || derived_vm_numvcpus == 0
+    (cpu_usage_rate_average * derived_vm_numvcpus) / 100.0
+  end
+end
+
+CustomAttribute
+class CustomAttribute
+  has_many   :metric_rollups, :as => :resource
+end
+
+TIME_RANGE = if opts[:days_given]
+               [opts[:days].days.ago.utc.beginning_of_hour..Time.now.utc.end_of_hour]
+             else
+               [opts[:hours].hours.ago.utc.beginning_of_hour..Time.now.utc.end_of_hour]
+             end
+
+labels = CustomAttribute.where(:section => ['labels', 'docker_labels'], :resource_type => ['ContainerImage', 'ContainerGroup'], :name => opts[:label])
+
+# Question - If a container is short lived, will we still be able to find it here?
+labels_containers = labels.each_with_object(Hash.new { |h,k| h[k] = [] }) { |l, h| h[l] += l.resource.containers.to_a }
+
+# Group metrics by label/project
+labels_metrics = Hash.new { |h,k| h[k] = [] }
+labels_containers.each do |l, containers|
+  containers.each do |c|
+    labels_metrics[[l, c.container_project]] += c.metrics.where(:timestamp => TIME_RANGE).to_a # This should be scoped to time range for the hour
+  end
+end
+
+# Calculate the max total (of all containers running with the same label/project) CPU cores for each hour of metrics
+labels_maxes = Hash.new { |h,k| h[k] = {} }
+labels_metrics.each do |label_project, metrics|
+  # Group the metrics for this label/project by timestamp so that we can determine the sum of CPU cores usage for
+  # each 20 interval and use that for comparing the max usages for the hour
+  metrics.group_by { |m| m.timestamp }.each do |ts, metrics|
+    hour_ts = ts.beginning_of_hour
+    cpu_cores = metrics.map { |m| m.derived_cpu_total_cores_used }.sum
+
+    labels_maxes[label_project][hour_ts] ||= { :cpu_usage_rate_average => 0 }
+    labels_maxes[label_project][hour_ts][:cpu_usage_rate_average] = cpu_cores if cpu_cores > labels_maxes[label_project][hour_ts][:cpu_usage_rate_average]
+  end
+end
+
+# Store the max CPU cores values in metric_rollups
+labels_maxes.each do |label_project, hours|
+  label, project = label_project
+  hours.each do |hour_ts, values|
+    mr = label.metric_rollups.find_or_initialize_by(
```

Comment:
hm, I see we group by label, object. Is that correct? 

the design says ```sum all usage by containers spawned from images with a {:name => "label_name", :value => "label_value"}  on 20 second intervals```

So with current grouping by label object, we have the group per 1 image, since the label object is copied per each image. I think we need to group just by ```{:name => "label_name", :value => "label_value"}``` ? The the group will be containers from all images with that label.

-------

That means we do not have the good object as a resource for metric_rollups though. Well, if we would have the tag mapping on, then the Tagging object would be what we want. Since there is 1 record for all labels ```{:name => "label_name", :value => "label_value"} ``` . Not sure if FINE ha working tag mapping (@cben?)

Or we could omit the resource and just store somewhere in the rollup what was label_value, label_name and project_name?

Reply:
(answering without context, PR TLDR...)
FINE has working tag mapping, but I'm not sure you should build on top of it :-)
user-visible considerations:
- user would have to define a mapping
- it slows down refresh a little bit
- it folds case and truncates to 50 char (because Classification table validation)
- it drops empty-value tags
code considerations:
- complexity
- tag mapping tests are already cumbersome, so building on top of it will make your tests cumbersome
- larger bug surface
- makes it hard to delete/redesign/refactor tag mapping to other approaches (like dynamic virtual tags that we comtemplated at some point)
- probably irrelevant if this is per ems per project, but: there is single Tag for (label, value) *per region*.  If you replicate multiple regions into a master region and run a global report there, you may see multiple identical Tags — defeating the benefit over CustomAttributes.